### ISSUE TITLE
fota not working with TizenRT Artik053s from samsung repository

### ISSUE THREAD

**Issue by pholat** [SamsungARTIK/u-boot-artik#3]
As in subject, issue description and steps to reproduce below:

Steps ( as in README.md ) 
* build u-boot 
* feed build bin with generated certs
* sign with pointed out filesign binary

Test 1:
* in TizenRT : 
   ` make && make download bl2 && make download os`
    * bl2 loads and doesn't start os ( due to bad signature ) - test success

Step:
* copy generated certs from step 2 to TizenRT for use for filesign binary in repo

Test 2:
* in TizenRT:
    `make download os`
    * bl2 loads and starts os - test success

Test 3:
* in TizenRT
    `make download ota`
    * bl2 loads, sees that there is ota img, fails on first `Magic` address check
    * it worked with bl2 binary from TizenRT 
    * test fail!

It seems for me that binary in TizenRT and build from branch from here is different.
bl2 form TizenRT was able to make ota update, while bl2 form repository does not.

Beside code not working fine: Would you be so kind to tell from what code bl2 in TizenRT is build?

**Comment by pholat** [comment]
I've checked and binary in TizenRT was made on 23.09.2017 hash: 59977f7 ,

While this one uses different ota file format - where could I find script/whatever to use new ota header format?

**Comment by JunhwanPark** [comment]
Hello @pholat 

* If you are using an S-module (ARTIK053S, ARTIK055S), you should also sign the bl2 binary. bl2 and os should use the same signing information. The updated bl2.bin file in Github uses the 'public key' provided by TizenRT. Please check the following link for that.
https://github.com/SamsungARTIK/u-boot-artik/tree/artik-05x#signing

* How did you make the ota binary? You need to create the OTA binary by pre-compiling the TizenRT binary with the necessary information for the OTA. We made the 'make image ota' command to simplify it.
In other words, do the following:
```bash
$ make                       // build
$ make image ota             // make ota.bin
$ make download ota          // fusing ota binary
```

**Comment by JunhwanPark** [comment]
Oh @pholat !
Is the Samsung Repository you are talking about 'https://github.com/Samsung/TizenRT'? This is the SamsungARTIK team. :) It may be different from original TizenRT. Try 'https://github.com/SamsungARTIK/TizenRT'.

The reason for putting magic information in OTA is that sudden power is not supplied in the middle of OTA, so that abnormal operation is not performed.

If you want to know about the changes to the OTA binary generation, check out the following links that I have compiled. 
https://gist.github.com/JunhwanPark/86829f0dce3f9c703c625df8dd216f30

This is a change in bash. In order to support OSX as well as Ubuntu, we are gradually changing the shell script to a python file. Perhaps you can check the following:
https://github.com/SamsungARTIK/TizenRT/blob/ARTIK-SW-18.05.00/build/configs/artik05x/artik05x_download.py

**Comment by pholat** [comment]
@JunhwanPark First of all: Thanks! 

It's right I'm using [TizenRT](https://github.com/Samsung/TizenRT), though I do my best to check changes in both repositories just in case, but I only do so on master which is why I didn't see the changes.

I really like introduced Magics as these have a lot of sense and the script "make image ota" was probably exactly what I needed :+1:  

As not the biggest fan of python for shell actions I could be biased, but shell scripts were a lot easier to read for me, I understand the need to make these, I just find them harder to read :)

All in all I'll test everything again and confirm everything is right so that you could close the issue.

**Comment by JunhwanPark** [comment]
@pholat Thank you!
If you have any questions about ARTIK05x and TizenRT, please feel free to call me.
